Ghidra/Processors/SuperH4/data/languages/SuperH4.ldefs

```diff
@@ -30,4 +30,19 @@
     <external_name tool="IDA-PRO" name="sh4"/>
     <external_name tool="gnu" name="sh4"/>
   </language>
+  <language processor="SuperH4"
```

Comment:
Thanks for the feedback. I would like clarify a few things for my own understanding before proceeding. 

1) According to http://shared-ptr.com/sh_insns.html, the SH-4 and SH-4A are very similar with the SH-4A support ~10 more instructions. The SH7091 is an SH-4 with two of the SH-4A instructions. 
2) From my understanding, by combining the SH-4 and SH-4A into a single processor module we are are incorrectly claiming SH-4 supports all of the additional instructions of SH-4A. As you mentioned the opcodes do not overlap so this should not affect disassembly/decompilation. However this would break tools built around processor modules for example instruction identification, processor verification, ROP gadget generation, etc. I see the **# arch    arch_sh_up** comment but I believe it is not used by Ghidra and is instead a hint to the developer. 
3) I claim SH-4 and SH-4A should be separate (or at least have instructions ifdef guarded out). I also claim the SH7091 is neither an SH-4 or an SH-4A and is correctly a variant of a the SH-4. 

I fully realize there is a maintenance cost for having additional processors. I want to weigh that against correctness. Ultimately it is your call and I am more than happy to go either way. Thanks again!